### PR TITLE
execute order optimisation

### PR DESCRIPTION
**Pull request by onemicky**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206754384322333

### CODE REVIEW

File: src/domain/synthetics/orders/simulateExecuteOrderTxn.tsx [hunk 1]
```diff
@@ -58,7 +69,7 @@ export async function simulateExecuteOrderTxn(chainId: number, p: SimulateExecut
         maxTimestamp: priceTimestamp,
       },
       ...(p.extraArgs ?? []),
-    ]),
+    ] as any),
```
Comment (hubert-de-lalye) [inline src/domain/synthetics/orders/simulateExecuteOrderTxn.tsx:72]: Not quite sure about this `any`s, as I'm a bit out of context with typechain. Is it possible to fix, or at least mitigate possible following issues if any?

Comment (onemicky) [reply src/domain/synthetics/orders/simulateExecuteOrderTxn.tsx]: Good point, thanks. Made types more strict

**Comment by onemicky** [comment]
Also added handy getters for all contracts we use:
```
  const dataStore = getDataStoreContract(chainId, provider);
  const multicall = getMulticallContract(chainId, provider);
  const exchangeRouter = getExchangeRouterContract(chainId, provider);
  ...
```

**Review by hubert-de-lalye — DISMISSED** [review]

File: src/config/contracts.ts [hunk 2]
```diff
@@ -314,3 +317,15 @@ export function getContract(chainId: number, name: string): string {
 
   return CONTRACTS[chainId][name];
 }
+
+function makeGetContract<T extends unknown & { abi: InterfaceAbi; connect: (address: string) => unknown }>(
```
Comment (hubert-de-lalye) [inline src/config/contracts.ts:321]:
```suggestion
function makeGetContract<T extends { abi: InterfaceAbi; connect: (address: string) => unknown }>(
```

`unknown & {...}` is equals to `{...}` as unknown a supertype of any type, so there's no necessity to use unknown

**Review by hubert-de-lalye — DISMISSED** [review]

**Review by hubert-de-lalye — APPROVED** [review]

**Pull request merged**